tests/slsa_analyzer/checks/test_build_as_code_check.py

```diff
@@ -1,149 +1,230 @@
-        maven_deploy.dynamic_data["ci_services"] = [ci_info]
-        assert check.run_check(maven_deploy, check_result) == CheckResultType.FAILED
-
-        # Use mvn but do not deploy artifacts.
-        no_maven_deploy = AnalyzeContext("use_build_tool", os.path.abspath("./"), MagicMock())
-        no_maven_deploy.dynamic_data["build_spec"]["tool"] = maven
-        bash_commands["commands"] = [["mvn", "verify"]]
-        no_maven_deploy.dynamic_data["ci_services"] = [ci_info]
-        assert check.run_check(no_maven_deploy, check_result) == CheckResultType.FAILED
-
-        # Use an invalid goal that has deploy keyword.
-        bash_commands["commands"] = [["mvnb", "deployblah"]]
-        no_maven_deploy.dynamic_data["ci_services"] = [ci_info]
-        assert check.run_check(no_maven_deploy, check_result) == CheckResultType.FAILED
-
-        # Use gradle to deploy the artifact.
-        gradle_deploy = AnalyzeContext("use_build_tool", os.path.abspath("./"), MagicMock())
-        gradle_deploy.dynamic_data["build_spec"]["tool"] = gradle
-        bash_commands["commands"] = [["./gradlew", "publishToSonatype"]]
-        gradle_deploy.dynamic_data["ci_services"] = [ci_info]
-        assert check.run_check(gradle_deploy, check_result) == CheckResultType.PASSED
-
-        # Test Jenkins.
-        maven_deploy = AnalyzeContext("use_build_tool", os.path.abspath("./"), MagicMock())
-        maven_deploy.dynamic_data["build_spec"]["tool"] = maven
-        ci_info["service"] = jenkins
-        bash_commands["commands"] = []
-        maven_deploy.dynamic_data["ci_services"] = [ci_info]
-        assert check.run_check(maven_deploy, check_result) == CheckResultType.FAILED
-
-        # Test Travis.
-        maven_deploy = AnalyzeContext("use_build_tool", os.path.abspath("./"), MagicMock())
-        maven_deploy.dynamic_data["build_spec"]["tool"] = maven
-        ci_info["service"] = travis
-        bash_commands["commands"] = []
-        maven_deploy.dynamic_data["ci_services"] = [ci_info]
-        assert check.run_check(maven_deploy, check_result) == CheckResultType.FAILED
-
-        # Test Circle CI.
-        maven_deploy = AnalyzeContext("use_build_tool", os.path.abspath("./"), MagicMock())
-        maven_deploy.dynamic_data["build_spec"]["tool"] = maven
-        ci_info["service"] = circle_ci
-        bash_commands["commands"] = []
-        maven_deploy.dynamic_data["ci_services"] = [ci_info]
-        assert check.run_check(maven_deploy, check_result) == CheckResultType.FAILED
-
-        # Test GitLab CI.
-        maven_deploy = AnalyzeContext("use_build_tool", os.path.abspath("./"), MagicMock())
-        maven_deploy.dynamic_data["build_spec"]["tool"] = maven
-        ci_info["service"] = gitlab_ci
-        bash_commands["commands"] = []
-        maven_deploy.dynamic_data["ci_services"] = [ci_info]
-        assert check.run_check(maven_deploy, check_result) == CheckResultType.FAILED
+
+@pytest.fixture()
```

Comment:
This `fixture` is not needed. Why not to directly call `BuildAsCodeCheck()` from the test function if nothing needs to be prepared before the test?

Reply:
Fixed in [29cdd3a](https://github.com/oracle-samples/macaron/pull/67/commits/29cdd3a9e361a0f5f29c835805a2d9f186b92d0a).